interfaces/udev/udev_test.go

```diff
@@ -0,0 +1,97 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package udev_test
+
+import (
+	"testing"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/ubuntu-core/snappy/interfaces/udev"
+	"github.com/ubuntu-core/snappy/testutil"
+)
+
+func Test(t *testing.T) {
+	TestingT(t)
+}
+
+type uDevSuite struct {
+	cmd *testutil.MockCmd
+}
+
+var _ = Suite(&uDevSuite{})
+
+func (s *uDevSuite) SetUpTest(c *C) {
+	s.cmd = testutil.MockCommand(c, "udevadm", 0)
+}
+
+func (s *uDevSuite) TearDownTest(c *C) {
+	s.cmd.Restore()
+}
+
+// Tests for ReloadRules()
+
+func (s *uDevSuite) TestReloadUDevRulesRunsUDevAdm(c *C) {
+	err := udev.ReloadRules()
+	c.Assert(err, IsNil)
+	c.Assert(s.cmd.Calls(), DeepEquals, []string{
+		"control --reload-rules",
+		"trigger",
+	})
+}
+
+func (s *uDevSuite) TestReloadUDevRulesReportsErrorsFromReloadRules(c *C) {
+	s.cmd.SetDynamicBehavior(1, func(n int) (string, int) {
```

Comment:
Sorry for bikesheding even further on this, but I don't understand what this test is doing. The helper, which is supposed to make things simple, is actually obscuring what's going on past the point of being understandable. This was supposed to be a trivial command, right?  exec, and this is what happens.. what is "dynamic behavior", 1, with n == 0 or 1 for a mocked command!?